### PR TITLE
Opinionated checkbox and radio layout

### PR DESCRIPTION
**Pull request by b-kelly**
Our suggested markup for checkboxes and radios is... not great. I've taken a more opinionated approach here and am opening up the PR for feedback. I also moved all the navigation entries for form components to their own section for easy access. Since we're only adding classes here, there are **no breaking changes** 🎉.

<details><summary>Markup comparison</summary>
Instead of this:

```html
<fieldset class="d-flex gs8 gsy fd-column">
    <legend class="flex--item s-label">Which types of fruit do you like? <span class="ml4 fw-normal fc-light">(Check all that apply)</span></legend>
    <div class="flex--item">
        <div class="d-flex gs16">
            <div class="flex--item">
                <div class="d-flex gs8 gsx">
                    <div class="flex--item">
                        <input class="s-checkbox" type="checkbox" name="" id="choice-horz-checkbox-1" />
                    </div>
                    <label class="flex--item s-label fw-normal" for="choice-horz-checkbox-1">Apples</label>
                </div>
            </div>
            <div class="flex--item">
                <div class="d-flex gs8 gsx">
                    <div class="flex--item">
                        <input class="s-checkbox" type="checkbox" name="" id="choice-horz-checkbox-2" />
                    </div>
                    <label class="flex--item s-label fw-normal" for="choice-horz-checkbox-2">Oranges</label>
                </div>
            </div>
            <div class="flex--item">
                <div class="d-flex gs8 gsx">
                    <div class="flex--item">
                        <input class="s-checkbox" type="checkbox" name="" id="choice-horz-checkbox-3" />
                    </div>
                    <label class="flex--item s-label fw-normal" for="choice-horz-checkbox-3">Bananas</label>
                </div>
            </div>
        </div>
    </div>
</fieldset>
```

we have this:

```html
<fieldset class="s-checkbox-group s-checkbox-group__horizontal">
    <legend class="s-label">Which types of fruit do you like? <span class="ml4 fw-normal fc-light">(Check all that apply)</span></legend>
    <div class="s-checkbox-control">
        <input class="s-checkbox" type="checkbox" name="" id="choice-horz-checkbox-1" />
        <label class="s-label" for="choice-horz-checkbox-1">Apples</label>
    </div>
    <div class="s-checkbox-control">
        <input class="s-checkbox" type="checkbox" name="" id="choice-horz-checkbox-2" />
        <label class="s-label" for="choice-horz-checkbox-2">Oranges</label>
    </div>
    <div class="s-checkbox-control">
        <input class="s-checkbox" type="checkbox" name="" id="choice-horz-checkbox-3" />
        <label class="s-label" for="choice-horz-checkbox-3">Bananas</label>
    </div>
</fieldset>
```

</details>

## Known issues
- [x] checkboxes/radios w/ descriptions are not aligned correctly
  image

## Follow-up
- [ ] add input groups / controls for other types of inputs?
- [ ] add an `.s-form` element that arranges all the input controls with consistent spacing, etc

### CODE REVIEW

File: docs/product/components/checkbox.html
Comment (dancormier) [inline docs/product/components/checkbox.html]: Naming things is hard, but I think `-control` is a good suffix for this.

File: lib/css/components/inputs.less
Comment (dancormier) [inline lib/css/components/inputs.less:293]:
```
.s-checkbox-group.s-checkbox-group__horizontal,
.s-checkbox-group.s-radio-group__horizontal,
.s-radio-group.s-checkbox-group__horizontal,
.s-radio-group.s-radio-group__horizontal {
    flex-direction: row;
}
```

This bugs me than it should, but I understand it

**Comment by dancormier** [comment]
@b-kelly I made a commit to resolve the _checkboxes/radios w/ descriptions_ issue. I _think_ it should be a fine resolution to the issue, but I could see something like different text rendering causing issues.

Feel free to revert/alter if it doesn't work in all cases or if you have something better.

<details>
<summary>In production</summary>

image
image

</details>

<details>
<summary>The fix</summary>

image

</details>

**Review by dancormier — APPROVED** [review]
LGTM!

I tested the previous markup to make sure it was still compatible and I added a markup deprecation message to the page.

<details>
<summary>Old markup example screenshot</summary>

image

</details>


Solid work here! @b-kelly I'll let you do the honors and merge this when it feels right 😎

**Comment by dancormier** [comment]
> I'll let you do the honors and merge this when it feels right 😎

I'm taking the honors! You snooze, you lose!

**Comment by b-kelly** [comment]
@dancormier Oops, I totally lost track of this and missed that message. Agreed: I snoozed and therefore, I losed. Thanks for the merge!

**Pull request merged**
